src/components/views/settings/Notifications.js

```diff
@@ -440,7 +442,7 @@ module.exports = React.createClass({
         if (needsUpdate.length > 0) {
             // If some of the rules need to be ported then wait for the porting
             // to happen and then fetch the rules again.
-            return q.allSettled(needsUpdate).then( function() {
+            return q.all(needsUpdate).then( function() {
```

Comment:
> You missed a 'q'. 

Oops.

> Also surely allSettled != all? Looks like you want reflect (which used to be settle)

the `reflect` stuff looks confusing to me, so I tried to avoid it, and instead added the `catch` handler at https://github.com/vector-im/riot-web/pull/4565/commits/1d2d086a58f6887f91c166f98e4314ebd2fa8c95#diff-beb48c1c7c8677b581a5b4177c817f28R434 to make the promises all succeed.